solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -967,7 +1036,7 @@ contract InterchainAccountRouter is Router {
         );
         emit RemoteCallDispatched(_destination, msg.sender, _router, _ism);
         return
-            mailbox.dispatch{value: msg.value}(
+            mailbox.dispatch{value: address(this).balance}(
```

Comment:
upon further consideration I think we should keep `msg.value` and omit the `receive`

Reply:
or alternatively specify the refund address as the `msg.sender` in hook metadata

Reply:
We now split the msg.value between the two dispatches.